wavelength-plugins/genrequests.js

```diff
@@ -0,0 +1,91 @@
+/********************************
+ * Pokemon Gen Requests for PS! *
+ * Created by Insist			*
+ ********************************/
+
+"use strict";
+
+function alertGenners(message) {
+	let genners = Db.genners.keys();
+	for (let u in genners) {
+		if (!Users(genners[u]) || !Users(genners[u]).connected) continue;
+		Users(genners[u]).send(`|pm|~Genner Alert|~|/raw ${message}`);
+	}
+	if (Rooms(`genrequests`)) {
+		Rooms(`genrequests`).add(`|c|~Genner Alert|/raw ${message}`).update();
+	}
+}
+
+exports.commands = {
+	genner: "genreq",
+	genners: "genreq",
+	genreqs: "genreq",
+	requestgen: "genreq",
+	genrequest: "genreq",
+	genreq: {
+		addgenner: "add",
+		addgen: "add",
+		approve: "add",
+		give: "add",
+		add: function (target, room, user) {
+			if (!this.can("lock")) return false;
+			if (!target || target.length > 18) return this.errorReply(`This command requires a target with a maximum of 18 characters.`);
```

Comment:
We should check `toId(target)` before checking `target.length` because a target could be something like `" a         b c d e      f g"`, which would be valid while condensed (which is what we want).

Reply:
ok